src/halmos/sevm.py

```diff
@@ -3405,11 +3339,19 @@ def finalize(ex: Exec):
 
                 elif opcode == EVM.CODECOPY:
                     loc: int = ex.mloc(check_size=False)
-                    offset: int = ex.int_of(ex.st.pop(), "symbolic CODECOPY offset")
+                    offset: int = ex.st.popi()
                     size: int = ex.int_of(ex.st.pop(), "symbolic CODECOPY size")
 
                     if size:
-                        codeslice: ByteVec = ex.pgm.slice(offset, size)
+                        # TODO: hide symbolic support behind a feature flag?
+                        # if the offset is symbolic, create a symbolic slice
+                        codeslice = (
+                            ex.pgm.slice(int(offset), size)
+                            if offset.is_concrete
+                            else ByteVec(
+                                BV(f"codeslice_uint{size * 8}_{uid()}", size=size * 8)
+                            )
+                        )
```

Comment:
is this adding a new feature or not? it would be better to have a new feature in a separate pr, if possible. actually, the usefulness of symbolic slice might need further discussion.